packages/webcomponentsjs/webcomponents-loader.js

```diff
@@ -157,16 +157,42 @@
   }
 
   if (polyfills.length) {
+    // When the Trusted Types API is available, `policy` is a
+    // `TrustedTypePolicy` with functions for creating trusted HTML, scripts,
+    // and script URLs. This policy is used below to (a) approve the bundle URL
+    // string created by the loader that is assigned to a `<script>`'s `src`
+    // attribute, (b) approve a constant script string that is assigned to that
+    // `<script>'s `onload` attribute, and (c) approve the string of HTML that
+    // the loader reads from that `<script>`'s `outerHTML`.
+    //
+    // If the Trusted Types API is not available, the returned object exposes a
+    // similar interface to a `TrustedTypePolicy`, but all of its functions are
+    // the identity function.
+    var policy = (() => {
+      var identity = function (x) {
+        return x;
+      };
+      var policyOptions = {
+        createHTML: identity,
+        createScript: identity,
+        createScriptURL: identity,
+      };
+      var policy =
+        window.trustedTypes &&
+        window.trustedTypes.createPolicy('webcomponents-loader', policyOptions);
+      return policy || policyOptions;
+    })();
+
     var url;
     var polyfillFile = 'bundles/webcomponents-' + polyfills.join('-') + '.js';
 
     // Load it from the right place.
     if (window.WebComponents.root) {
-      url = window.WebComponents.root + polyfillFile;
+      url = policy.createScriptURL(window.WebComponents.root + polyfillFile);
```

Comment:
This doesn't check that Webcomponents.root is a TrustedScriptUrl, but the README suggested that it would

If we want to check that, I think the code would be roughly:

```javascript
url = window.WebComponents.root + polyfillFile;
if (window.trustedTypes && 
    window.trustedTypes.isScriptURL(window.WebComponents.root)) {
  url = policy.createScriptURL(url);
}
```

Reply:
It doesn't because it would be a breaking change to always check if it were one if `window.trustedTypes` exists. However, @koto [says it's considered an antipattern to detect whether or not trusted types are being enforced](https://github.com/webcomponents/polyfills/pull/501#discussion_r846735919), which would be necessary to propagate both the requirement that `WebComponents.root` be a `TrustedScriptURL` without throwing an error if trusted types are being reported, but not enforced.

Reply:
If we instead only make url a TrustedScriptURL if WebComponents.root is a TrustedScriptUrl then existing users get the current behavior, and that's only a problem if they've opted into trusted types.

But if they've opted into trusted types, then they _want_ to be told about potential security issues, and so if they're using window.WebComponents.root then they _want to know_ that it's a potential security issue, and it should only be set to a trusted script url.

Reply:
Ok, I now buy that this both isn't a breaking change and propagates safety since they can't use the loader with trusted types enabled currently.

Reply:
Updated.